Fix similar_objects() in multi inheritance contexts (KeyError)

czr137 opened the pull request:
This addresses issues wagtail users have had using `similar_objects()` with `ClusterTaggableManager`

This PR would resolve both #424 and #508 and close issue #80.

It is an updated and rebased version of the commits made in #424 by nickhudkins.

It finishes the work in PR #424 in that drops the workarounds for older versions of Django as requested by @jdufresne in #424.

czr137 commented:
At first glance it looks like a problem with the test scripts for 3.6 and 3.7 and not an issue with this PR. Can a JazzBand maintainer confirm?

czr137 commented:
Test scripts for 3.6 and 3.7 are attempting to use a version of Django that no longer supports python 3.6 and 3.7. Looks like these test builds could be dropped, or they need to be edited to specify an older version of Django.

auvipy commented on tests/models.py:
```diff
@@ -57,6 +57,29 @@ def __str__(self):
+    )
+
+    class Meta:
+        unique_together = [["content_object", "tag"]]
```
instead of unique_together you could try UniqueContraint

auvipy replied:
https://docs.djangoproject.com/en/2.2/ref/models/options/#unique-together

czr137 replied:
There are about a half dozen occurrences of unique_together in this file, and none of them have been switched over to UniqueConstraint yet. I think for clarity, as this would be an unrelated change, it would be better to change all the unique-together to UniqueConstraint with its own issue / PR.

czr137 replied:
UniqueConstraint was introduced in Django 2.2, so if I'm not mistaken, this would also mean the min version of Django would have to be bumped to 2.2 and version 1 support would have to be dropped. This is probably fine at this point, but it is also beyond the scope of this PR.

czr137 commented:
Django 4.0 has dropped support for python 3.6,3.7.
As such, I modified the tox test script so that it doesn't test python 3.6 and 3.7 against django master, which has been updated to support >=3.8.

auvipy commented on tests/migrations/0001_initial.py:
```diff
@@ -31,6 +31,21 @@ class Migration(migrations.Migration):
                 ("title", models.CharField(max_length=100)),
             ],
         ),
+        migrations.CreateModel(
```
did you created the migrations yourselves? are they going to break the downward migrations?

czr137 replied:
> did you created the migrations yourselves?

Yes.

> are they going to break the downward migrations?

Please review them. The tests are passing.

czr137 commented:
@auvipy, please let me know if there is anything else I can do to help get this fix in. If there is some other way to demonstrate the migrations I added are correct, let me know and I'll try to do that over the weekend.

czr137 commented:
@auvipy, I removed my manual migrations and added a migrations file with django's generated migration. I rebased onto @jezdez test updates so the two PR's are consitent.
Tests are all passing and test migrations are fixed.

czr137 commented:
#599 can also be closed when this is merged in.

jdufresne commented:
@auvipy I'm no longer actively using or maintaining django-taggit. Please feel free to review, merge, and release as you see fit.
